File: Argon2id/Argon2id.php
<?php
// ### the minimal parameter set is probably UNSECURE ###
function generateArgon2idMinimal($password, $salt) {
    $opsLimit= 2;
    $memLimit = 8192 * 1024;
    $outputLength =  32;
    return sodium_crypto_pwhash ($outputLength, $password, $salt,   $opsLimit, $memLimit, SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13);
}

function generateArgon2idInteractive($password, $salt) {
    $opsLimit= 2;
    $memLimit = 66536 * 1024;
    $outputLength =  32;
    return sodium_crypto_pwhash ($outputLength, $password, $salt,   $opsLimit, $memLimit, SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13);
}

function generateArgon2idModerate($password, $salt) {
    $opsLimit= 3;
    $memLimit = 262144 * 1024;
    $outputLength =  32;
    return sodium_crypto_pwhash ($outputLength, $password, $salt,   $opsLimit, $memLimit, SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13);
}

function generateArgon2idSensitive($password, $salt) {
    $opsLimit= 4;
    $memLimit = 1048576 * 1024;
    $outputLength =  32;
    return sodium_crypto_pwhash ($outputLength, $password, $salt,   $opsLimit, $memLimit, SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13);
}

function base64Encoding($input)
{
    return base64_encode($input);
}

function generateSalt16Byte()
{
    return openssl_random_pseudo_bytes(16, $crypto_strong);
}

function generateFixedSalt16Byte()
{
    // ### security warning - never use this in production ###
    return "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00";
}

echo 'Generate a 32 byte long encryption key with Argon2id' . PHP_EOL;

$password = "secret password";
echo 'password: ' . $password . PHP_EOL;

// ### security warning - never use a fixed salt in production, this is for compare reasons only
//$salt = generateFixedSalt32Byte();
$salt = generateFixedSalt16Byte();
// please use below generateSalt16Byte()
//$salt = generateSalt16Byte();
echo 'salt (Base64): ' . base64Encoding($salt) . PHP_EOL;

// ### the minimal parameter set is probably UNSECURE ###
$encryptionKeyArgon2id = generateArgon2idMinimal($password, $salt);
echo 'encryptionKeyArgon2id (Base64) minimal:     ' . base64Encoding($encryptionKeyArgon2id) . PHP_EOL;

$encryptionKeyArgon2id = generateArgon2idInteractive($password, $salt);
echo 'encryptionKeyArgon2id (Base64) interactive: ' . base64Encoding($encryptionKeyArgon2id) . PHP_EOL;

$encryptionKeyArgon2id = generateArgon2idModerate($password, $salt);
echo 'encryptionKeyArgon2id (Base64) moderate:    ' . base64Encoding($encryptionKeyArgon2id) . PHP_EOL;

$encryptionKeyArgon2id = generateArgon2idSensitive($password, $salt);
echo 'encryptionKeyArgon2id (Base64) sensitive:   ' . base64Encoding($encryptionKeyArgon2id) . PHP_EOL;
?>

File: Argon2id/Argon2id.java
package Argon2id;

import org.bouncycastle.crypto.generators.Argon2BytesGenerator;
import org.bouncycastle.crypto.params.Argon2Parameters;

import java.nio.charset.StandardCharsets;
import java.security.SecureRandom;
import java.util.Base64;

public class Argon2id {
    public static void main(String[] args) {
        // uses Bouncy Castle
        System.out.println("Generate a 32 byte long encryption key with Argon2id");

        String password = "secret password";
        System.out.println("password: " + password);

        // ### security warning - never use a fixed salt in production, this is for compare reasons only
        byte[] salt = generateFixedSalt16Byte();
        // please use below generateSalt16Byte()
        //byte[] salt = generateSalt16Byte();
        System.out.println("salt (Base64): " + base64Encoding(salt));

        // ### the minimal parameter set is probably UNSECURE ###
        String encryptionKeyArgon2id = base64Encoding(generateArgon2idMinimal(password, salt));
        System.out.println("encryptionKeyArgon2id (Base64) minimal:     " + encryptionKeyArgon2id);

        encryptionKeyArgon2id = base64Encoding(generateArgon2idInteractive(password, salt));
        System.out.println("encryptionKeyArgon2id (Base64) interactive: " + encryptionKeyArgon2id);

        encryptionKeyArgon2id = base64Encoding(generateArgon2idModerate(password, salt));
        System.out.println("encryptionKeyArgon2id (Base64) moderate:    " + encryptionKeyArgon2id);

        encryptionKeyArgon2id = base64Encoding(generateArgon2idSensitive(password, salt));
        System.out.println("encryptionKeyArgon2id (Base64) sensitive:   " + encryptionKeyArgon2id);
    }

    // ### the minimal parameter set is probably UNSECURE ###
    public static byte[] generateArgon2idMinimal(String password, byte[] salt) {
        int opsLimit = 2;
        int memLimit = 8192;
        int outputLength = 32;
        int parallelism = 1;
        Argon2Parameters.Builder builder = new Argon2Parameters.Builder(Argon2Parameters.ARGON2_id)
                .withVersion(Argon2Parameters.ARGON2_VERSION_13) // 19
                .withIterations(opsLimit)
                .withMemoryAsKB(memLimit)
                .withParallelism(parallelism)
                .withSalt(salt);
        Argon2BytesGenerator gen = new Argon2BytesGenerator();
        gen.init(builder.build());
        byte[] result = new byte[outputLength];
        gen.generateBytes(password.getBytes(StandardCharsets.UTF_8), result, 0, result.length);
        return result;
    }

    public static byte[] generateArgon2idInteractive(String password, byte[] salt) {
        int opsLimit = 2;
        int memLimit = 66536;
        int outputLength = 32;
        int parallelism = 1;
        Argon2Parameters.Builder builder = new Argon2Parameters.Builder(Argon2Parameters.ARGON2_id)
                .withVersion(Argon2Parameters.ARGON2_VERSION_13) // 19
                .withIterations(opsLimit)
                .withMemoryAsKB(memLimit)
                .withParallelism(parallelism)
                .withSalt(salt);
        Argon2BytesGenerator gen = new Argon2BytesGenerator();
        gen.init(builder.build());
        byte[] result = new byte[outputLength];
        gen.generateBytes(password.getBytes(StandardCharsets.UTF_8), result, 0, result.length);
        return result;
    }

    public static byte[] generateArgon2idModerate(String password, byte[] salt) {
        int opsLimit = 3;
        int memLimit = 262144;
        int outputLength = 32;
        int parallelism = 1;
        Argon2Parameters.Builder builder = new Argon2Parameters.Builder(Argon2Parameters.ARGON2_id)
                .withVersion(Argon2Parameters.ARGON2_VERSION_13) // 19
                .withIterations(opsLimit)
                .withMemoryAsKB(memLimit)
                .withParallelism(parallelism)
                .withSalt(salt);
        Argon2BytesGenerator gen = new Argon2BytesGenerator();
        gen.init(builder.build());
        byte[] result = new byte[outputLength];
        gen.generateBytes(password.getBytes(StandardCharsets.UTF_8), result, 0, result.length);
        return result;
    }

    public static byte[] generateArgon2idSensitive(String password, byte[] salt) {
        int opsLimit = 4;
        int memLimit = 1048576;
        int outputLength = 32;
        int parallelism = 1;
        Argon2Parameters.Builder builder = new Argon2Parameters.Builder(Argon2Parameters.ARGON2_id)
                .withVersion(Argon2Parameters.ARGON2_VERSION_13) // 19
                .withIterations(opsLimit)
                .withMemoryAsKB(memLimit)
                .withParallelism(parallelism)
                .withSalt(salt);
        Argon2BytesGenerator gen = new Argon2BytesGenerator();
        gen.init(builder.build());
        byte[] result = new byte[outputLength];
        gen.generateBytes(password.getBytes(StandardCharsets.UTF_8), result, 0, result.length);
        return result;
    }

    private static byte[] generateSalt16Byte() {
        SecureRandom secureRandom = new SecureRandom();
        byte[] salt = new byte[16];
        secureRandom.nextBytes(salt);
        return salt;
    }

    private static byte[] generateFixedSalt16Byte() {
        // ### security warning - never use this in production ###
        byte[] salt = new byte[16]; // 16 x0's
        return salt;
    }

    private static String base64Encoding(byte[] input) {
        return Base64.getEncoder().encodeToString(input);
    }
}
